libnavigation-base/src/main/java/com/mapbox/navigation/base/options/OnboardRouterEndpointOptions.kt

```diff
@@ -0,0 +1,72 @@
+package com.mapbox.navigation.base.options
+
+import android.content.Context
+import java.io.File
+import java.net.URI
+
+/**
+ * Defines endpoint's properties to retrieve routing tiles.
+ *
+ * @param tilesUri tiles endpoint
+ * @param version version of tiles
+ * @param userAgent HttpClient UserAgent
+ * @param builder used for updating options
+ */
+data class OnboardRouterEndpointOptions(
+    val tilesUri: URI,
+    val version: String,
+    val userAgent: String,
+    val builder: Builder
+) {
+    /**
+     * @return the builder that created the [OnboardRouterEndpointOptions]
+     */
+    fun toBuilder() = builder
+
+    /**
+     * Creates a path to store the road network tiles.
+     */
+    fun createDefaultTilePath(context: Context): String {
+        val directoryVersion = "Offline/${tilesUri.host}/$version/tiles"
+        return File(context.filesDir, directoryVersion).absolutePath
+    }
```

Comment:
This returns an internal storage path, should we expose this or let the client decide what path to use? It can be also an external storage path 🤔

Reply:
This is the default, open question for what exactly the default should be. I don't have a clear winner for Internal vs external. Seems like an "option" 🙃 . Could update this function. I propose addressing that in a later change so we are testing the differences though

Reply:
My point is if we should expose this from here 🤔 Before we were adding the default in `MapboxNavigation` if no config was provided. Wondering if we should even expose it 🤔

Reply:
I think we should provide a default. Otherwise those who use it will copy and paste our versions.

Also this is only moving it from MapboxNavigation, into OnboardRouterEndpointOptions. This is at least easier to implement as someone who uses it

Reply:
But I'm not sure if you mean, it should not live in "OnboardRouterEndpointOptions". that is a fair concern, the function could depend on context and options, and live somewhere else

``` kotlin
    fun createDefaultTilePath(context: Context, OnboardRouterEndpointOptions options): String {
        val directoryVersion = "Offline/${tilesUri.host}/$version/tiles"
        return File(context.filesDir, directoryVersion).absolutePath
    }
```

Reply:
another issue providing the default. is if we change it, it'll change it for those downstream. but again, that exists today. so if this is broken, it's not breaking it any more or less right? :)

Reply:
Maybe the concern is exposing the path? Also a fair thought. We could have storage options (internal, external, size, path is predetermined)

All these things are good ideas. But out of scope of this change. This change is to use a Builder pattern without changing functionality

Reply:
OK removed all the things that are confusing to me :)

https://github.com/mapbox/mapbox-navigation-android/pull/3020/commits/c7e983b44b080ad8783227de81e40f17ad78dd3f

Reply:
removing the context from the builder. same issue exists for the MapboxDistanceFormatter, why should you have to pass the application context to everything? seems like you should pass it once and the sdk will remember it